charts/incubator/misskey/values.yaml

```diff
@@ -0,0 +1,159 @@
+image:
+  repository: misskey/misskey
+  pullPolicy: IfNotPresent
+  tag: 12.119.0@sha256:a7beecb59b5089068463ca3847ac34dcb8445108da75ed8f56822f670053aa31
+
+
+securityContext:
+  readOnlyRootFilesystem: false
+  runAsNonRoot: false
+
+podSecurityContext:
+  runAsUser: 0
+  runAsGroup: 0
+  fsGroup: 33
+
+
+service:
+  main:
+    ports:
+      main:
+        port: 3000
+        targetPort: 3000
+
+hostAliases:
+  - ip: '{{ .Values.env.AccessIP | default "127.0.0.1" }}'
+    hostnames:
+      - "{{ if .Values.ingress.main.enabled }}{{ with (first .Values.ingress.main.hosts) }}{{ .host }}{{ end }}{{ else }}placeholder.fakedomain.dns{{ end }}"
+
+
+initContainers:
+  prestart:
+    image: '{{ include "tc.common.images.selector" . }}'
+    securityContext:
+      runAsUser: 0
+      runAsGroup: 0
+      privileged: true
+    env:
+      # IP used for exposing misskey
+      # Often the service or nodePort IP
+      # Defaults to the main serviceName for CI purposes.
+      - name: POSTGRES_PASSWORD
+        valueFrom:
+          secretKeyRef:
+            name: dbcreds
+            key: postgresql-password
+      - name: POSTGRES_HOST
+        valueFrom:
+          secretKeyRef:
+            name: dbcreds
+            key: plainporthost
+      - name: REDIS_HOST
+        valueFrom:
+          secretKeyRef:
+            name: rediscreds
+            key: plainhost
+      - name: REDIS_HOST_PASSWORD
+        valueFrom:
+          secretKeyRef:
+            name: rediscreds
+            key: redis-password
+    # Modify mount permissions and inject postgres/redis secrets into config file
+    command:
+      - "/bin/sh"
+      - "-c"
+      - |
+        /bin/bash <<'EOF'
+        echo "Forcing permissions on userdata folder..."
+        if nfs4xdr_getfacl && nfs4xdr_getfacl | grep -qv "Failed to get NFSv4 ACL"; then
+          echo "NFSv4 ACLs detected, Trying to override permissions using nfs4_setfacl..."
+          nfs4_setfacl -R -a A:g:33:RWX "/misskey/files"
+        else
+          echo "No NFSv4 ACLs detected, trying to override permissions using chown/chmod..."
+          echo "checking ownership..."
+          if [ $(stat -c %g .) -eq 33 ]; then
+            echo "Ownership already set to 33, skipping..."
+          else
+            echo "Changing ownership to group 33..."
+            chown -R :33 "/misskey/files"
+          fi
+          chmod 770 /misskey/files
+        fi
+        cp /var/misskey/.config/default.yml /misskey/.config/
+        sed -i "s/POSTGRES_HOST/$POSTGRES_HOST/" /misskey/.config/default.yml
+        sed -i "s/POSTGRES_PASSWORD/$POSTGRES_PASSWORD/" /misskey/.config/default.yml
+        sed -i "s/REDIS_HOST/$REDIS_HOST/" /misskey/.config/default.yml
+        sed -i "s/REDIS_PASSWORD/$REDIS_HOST_PASSWORD/" /misskey/.config/default.yml
```

Comment:
Why don't you prepare the configmap directly and just mount it in the container?
There is no need for sed here.

Reply:
Upstream Misskey currently reads in password and host data through the config map yaml file. Since as far as I know, to protect secrets, Kubernetes does not allow you to embed raw secrets in the deployment files. To get them in the config file, I used an init container to read the generated secrets from the env variables and write to the mounted config file. Open to suggestions if there is a better method you are aware of.

Reply:
You can pull values from dependencies like redis/pg and put them in the configmap.
Search for `.Values.redis.redisPassword` for examples

You can also consider put the whole config in a secret, instead of a configmap

Reply:
I found redisPassword and postgresqlPassword. For the host names, is <appname>-redis and <appname>-postgresql considered stable, or is the host name generated using a different method?

Reply:
Yes those are stable, you can use this:
`{{ printf "%v-%v" .Release.Name "postgresql" }}` 
`{{ printf "%v-%v" .Release.Name "redis" }}`